src/components/new-safe/steps/Step3/index.tsx

```diff
@@ -0,0 +1,166 @@
+    threshold: safeThreshold,
+    saltNonce,
+  })
+
+  const totalFee =
+    gasLimit && maxFeePerGas && maxPriorityFeePerGas
+      ? formatVisualAmount(maxFeePerGas.add(maxPriorityFeePerGas).mul(gasLimit), chain?.nativeCurrency.decimals)
+      : '> 0.001'
+
+  const formMethods = useForm<CreateSafeStep3Form>({
+    mode: 'all',
+    defaultValues: {
+      [CreateSafeStep3Fields.name]: newSafeName,
+      [CreateSafeStep3Fields.owners]: safeOwners,
+      [CreateSafeStep3Fields.threshold]: safeThreshold,
+    },
+  })
+
+  const { handleSubmit } = formMethods
+
+  const onSubmit = (data: CreateSafeStep3Form) => {
+    console.log(data)
+  }
+
+  return (
+    <StepCard
+      step={3}
+      title="Review"
+      subheader={`You're about to create a new Safe on ${chain?.chainName} and will have to confirm a transaction with your currently connected wallet.`}
+      content={
+        <form onSubmit={handleSubmit(onSubmit)} id={STEP_3_FORM_ID}>
+          <FormProvider {...formMethods}>
+            <Grid container spacing={3}>
+              <Grid item>
+                <Grid container spacing={3}>
+                  <ReviewRow name="Network" value={<ChainIndicator chainId={chain?.chainId} inline />} />
+                  <ReviewRow name="Name" value={<Typography>{newSafeName}</Typography>} />
+                  <ReviewRow
+                    name="Owners"
+                    value={
+                      <Box className={css.ownersArray}>
+                        {safeOwners.map((owner, index) => (
+                          <EthHashInfo address={owner.address} showPrefix={false} key={index} />
+                        ))}
+                      </Box>
+                    }
+                  />
+                  <ReviewRow
+                    name="Threshold"
+                    value={
+                      <Typography>
+                        {safeThreshold} out of {safeOwners.length} owner(s)
+                      </Typography>
+                    }
+                  />
+                </Grid>
+              </Grid>
+
+              <Grid item xs={12}>
+                <Divider sx={{ ml: '-52px', mr: '-52px', mb: 4, mt: 3 }} />
+                <Grid item xs={12}>
+                  <Grid container spacing={3}>
+                    <ReviewRow
+                      name="Est. network fee"
+                      value={
+                        <Typography variant="body1">
+                          <b>
+                            {totalFee} {chain?.nativeCurrency.symbol}
+                          </b>
+                        </Typography>
+                      }
+                    />
+                    <ReviewRow
+                      name=""
+                      value={
+                        <Typography color="text.secondary">
+                          You will have to confirm a transaction with your currently connected wallet.
+                        </Typography>
```

Comment:
This should look like this:
<img width="432" alt="Screenshot 2022-10-27 at 15 48 39" src="https://user-images.githubusercontent.com/381895/198302534-b8970860-dd88-49c7-a583-8f6678cf1f43.png">